core/src/main/java/nl/nn/adapterframework/jdbc/JdbcTransactionalStorage.java

```diff
@@ -706,71 +705,72 @@ private String createResultString(boolean isMessageDifferent){
 
 	@Override
 	public String storeMessage(String messageId, String correlationId, Date receivedDate, String comments, String label, S message) throws SenderException {
-		IbisTransaction itx = null;
-		if (txManager!=null) {
+		if (messageId == null) {
+			throw new SenderException("messageId cannot be null");
+		}
+		if (correlationId == null) {
+			throw new SenderException("correlationId cannot be null");
+		}
+
+		IbisTransaction itx;
+		if (txManager != null) {
 			itx = new IbisTransaction(txManager, txDef, ClassUtils.nameOf(this));
+		} else {
+			itx = null;
 		}
 		try {
-			String result;
-			if (messageId==null) {
-				throw new SenderException("messageId cannot be null");
-			}
-			if (correlationId==null) {
-				throw new SenderException("correlationId cannot be null");
-			}
 			try (Connection conn = getConnection()) {
-				Timestamp receivedDateTime = new Timestamp(receivedDate.getTime());
-				if (messageId.length()>MAXIDLEN) {
-					messageId=messageId.substring(0,MAXIDLEN);
-				}
-				if (correlationId.length()>MAXCIDLEN) {
-					correlationId=correlationId.substring(0,MAXCIDLEN);
-				}
-				if (comments!=null && comments.length()>MAXCOMMENTLEN) {
-					comments=comments.substring(0,MAXCOMMENTLEN);
-				}
-				if (label!=null && label.length()>MAXLABELLEN) {
-					label=label.substring(0,MAXLABELLEN);
-				}
-				result = storeMessageInDatabase(conn, messageId, correlationId, receivedDateTime, comments, label, message);
-				if (result==null) {
-					result=retrieveKey(conn,messageId,correlationId,receivedDateTime);
+				return storeMessage(conn, messageId, correlationId, receivedDate, comments, label, message);
+			} catch (SenderException e) {
+				if (itx != null) {
```

Comment:
Misschien is het op dit moment makkelijker om een aparte functie te maken voor het `if (itx != null)` probleem.